### PR TITLE
Add bottom sheet sign to all txns

### PR DESCRIPTION
**Pull request by Perronef5**
## CHANGES 📲
- Add bottom sheet sign provider to all app interactions

## DEMO 📸
Simulator Screen Shot - iPhone Xʀ - 2023-05-12 at 17 33 37

## TODO 📝
- [x] Fix issue where simulate txn is showing `Send DNT TOKEN` for claim rewards instead of `Receive DNT TOKEN`
- [x] Localize bottom sheet provider additional messages
- [x] Show some sort of txn estimation for claim all rewards bottom sheet

### CODE REVIEW

File: src/features/collectables/TransferCollectableScreen.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ const TransferCollectableScreen = () => {
   const { anchorProvider } = useSolana()
   const addressBookRef = useRef<AddressBookRef>(null)
   const colors = useColors()
-  const { showOKCancelAlert } = useAlert()
+  const [transfering, setTransfering] = useState(false)
```
Comment (tyler-whitman) [inline src/features/collectables/TransferCollectableScreen.tsx:77]: typo here: transferring

File: src/store/slices/solanaSlice.ts [hunk 2]
```diff
@@ -158,49 +126,25 @@ export const makePayment = createAsyncThunk(
...
+    { account, transferTxn, cluster, anchorProvider }: CollectablePaymentInput,
     { dispatch },
   ) => {
     if (!account?.solanaAddress) throw new Error('No solana account found')
 
-    const compressedNFT = collectable as CompressedNFT
-    const nft = collectable as Collectable
-
     try {
-      const transfer = compressedNFT?.compression?.compressed
```
Comment (tyler-whitman) [inline src/store/slices/solanaSlice.ts:176]: Why did all of this get removed?

**Review by tyler-whitman — APPROVED** [review]

File: src/store/slices/solanaSlice.ts (hunk 2 above)
Comment (Perronef5) [reply src/store/slices/solanaSlice.ts:176]: Transaction creation got moved to the `useSubmitTxn` hook in order to simulate it with the bottom sheet

File: src/hooks/useSubmitTxn.ts [hunk 3]
```diff
@@ -37,70 +45,204 @@ export default () => {
...
+        currentAccount.solanaAddress,
+        currentAccount.address,
+        payments,
+        mintAddress,
+      )
+
+      const serializedTx = paymentTxn.serialize()
+      setSerializedTx(Buffer.from(serializedTx))
+
+      // wait 0.5 second to allow the bottom sheet to load txn
```
Comment (ChewingGlass) [inline src/hooks/useSubmitTxn.ts:72]: Is this worrisome at all, on a slow connection?

File: src/hooks/useSubmitTxn.ts [hunk 4]
```diff
@@ -122,6 +271,29 @@ export default () => {
...
       }
 
+      if (!walletSignBottomSheetRef) {
+        throw new Error('No wallet sign bottom sheet ref')
+      }
+
+      const serializedTx = txns[0].serialize({
+        requireAllSignatures: false,
+      })
+      setSerializedTx(serializedTx)
```
Comment (ChewingGlass) [inline src/hooks/useSubmitTxn.ts:281]: It's always only setting one tx throughout this file. What if there are multiple?

File: src/hooks/useSubmitTxn.ts [hunk 5]
```diff
@@ -131,22 +303,50 @@ export default () => {
...
         throw new Error(t('errors.account'))
       }
 
       if (!currentAccount) {
         throw new Error(t('errors.account'))
       }
 
+      // Estimating fee in lamports. Not the best but works for now (:
+      const claimAllEstimatedFee =
+        hotspots.length !== 0 ? (hotspots.length / 2) * 5000 : 5000
```
Comment (ChewingGlass) [inline src/hooks/useSubmitTxn.ts:333]: There's two signers on each claim tx. That's 10k lamports each.

Comment (ChewingGlass) [reply src/hooks/useSubmitTxn.ts:333]: on solana it's 5000 lamports per signer

**Pull request merged**